### PR TITLE
API: Improper x/y arg given to df.plot

### LINKED ISSUE #18671: UserWarning about columns and attribute while plotting in Jupyter
#### Code Sample, a copy-pastable example if possible

```python
vibe_df.plot(x='Time', y='Fz')     # This works just fine.
vibe_df.plot(x='Time', y=['Fz'])     # This raises a UserWarning:

```
#### Problem description

I get a UserWarning while plotting a DataFrame.  It only occurs if I pass a list of columns for y.

UserWarning is:
C:\Program Files\Anaconda3\lib\site-packages\pandas\plotting\_core.py:1714: UserWarning: Pandas doesn't allow columns to be created via a new attribute name - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access
  series.name = label

#### Expected Output
<just the graph>

#### Output of ``pd.show_versions()``
<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.4.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.21.0
pytest: 2.9.2
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.24.1
numpy: 1.11.3
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.1.0
tables: 3.2.2
numexpr: 2.6.4
feather: None
matplotlib: 2.1.0
openpyxl: 2.3.2
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: 3.6.4
bs4: 4.5.1
html5lib: None
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by masongallo**
- [x] closes #18671 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry 

I'm not incredibly familiar with your codebase so I did my best to follow conventions. Not sure what to do with whatsnew for this case?

Description:

- Validation of x or y arg to `df.plot` to match specifications in documentation

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Thanks, this will be for 0.22 since it's an API change. Some people may have been ignoring the warning and relying on this.

Did you look into how difficult it'd be to properly support multiple columns for `y`? At the very least, we would need to make sure that `label` was a a list-like of the same length as `y`. Not sure what else.

**Comment by MasonGallo** [comment]
> Did you look into how difficult it'd be to properly support multiple columns for y?

I did think about this - the usecase that isn't already covered wasn't clear to me, especially given the complexity it would add to the method/API. The `plot` method already has `subplots` and `secondary_y` for example. What do you think?

**Comment by TomAugspurger** [comment]
I don't have a strong opinion. I think the equivalent output would be `df.set_index("x")[y].plot()`, which isn't so bad. It's probably best to just assert that `y` is a a single column, so your changes here look good.

The release not can go in `doc/source/whatsnew/v0.22.0.txt` under plotting.

**Comment by MasonGallo** [comment]
> The release not can go in doc/source/whatsnew/v0.22.0.txt under plotting.

Thanks, I see api changes section will update that

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -2170,6 +2170,15 @@ def test_invalid_kind(self):
         with pytest.raises(ValueError):
             df.plot(kind='aasdf')
 
+    def test_invalid_xy_args(self):
+        df = DataFrame({"A": [1, 2], 'B': [3, 4], 'C': [5, 6]})
```
Comment (jreback) [inline pandas/tests/plotting/test_frame.py]: can yu add the issue number here

File: pandas/tests/plotting/test_frame.py [hunk 2]
```diff
@@ -2170,6 +2170,15 @@ def test_invalid_kind(self):
         with pytest.raises(ValueError):
             df.plot(kind='aasdf')
 
+    def test_invalid_xy_args(self):
```
Comment (jreback) [inline pandas/tests/plotting/test_frame.py]: can you parameterize this on x, y

File: pandas/tests/plotting/test_frame.py [hunk 3]
```diff
@@ -2170,6 +2170,15 @@ def test_invalid_kind(self):
...
+        df = DataFrame({"A": [1, 2], 'B': [3, 4], 'C': [5, 6]})
+        bad_arg = ['B', 'C']
+        valid_arg = 'A'
+        with pytest.raises(ValueError):
```
Comment (jreback) [inline pandas/tests/plotting/test_frame.py]: can you also add a dataframe with duplicate columns as a test, e.g.

```
In [6]: df = DataFrame([[1, 3, 5], [2, 4, 6]], columns=list('AAB'))

In [7]: df
Out[7]: 
   A  A  B
0  1  3  5
1  2  4  6

In [8]: df['A']
Out[8]: 
   A  A
0  1  3
1  2  4
```

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.22.0.txt [hunk 4]
```diff
@@ -188,6 +188,7 @@ Other API Changes
 - :func:`pandas.DataFrame.merge` no longer casts a ``float`` column to ``object`` when merging on ``int`` and ``float`` columns (:issue:`16572`)
 - The default NA value for :class:`UInt64Index` has changed from 0 to ``NaN``, which impacts methods that mask with NA, such as ``UInt64Index.where()`` (:issue:`18398`)
 - Refactored ``setup.py`` to use ``find_packages`` instead of explicitly listing out all subpackages (:issue:`18535`)
+- :func: `DataFrame.plot` now raises a ``ValueError`` when the ``x`` or ``y`` argument is improperly formed (:issue:`18671`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: you can move this to plotting bugs.

File: pandas/plotting/_core.py [hunk 5]
```diff
@@ -1706,11 +1706,15 @@ def _plot(data, x=None, y=None, subplots=False,
             if x is not None:
                 if is_integer(x) and not data.columns.holds_integer():
                     x = data.columns[x]
+                elif not isinstance(data[x], Series):
```
Comment (jreback) [inline pandas/plotting/_core.py]:
```
from pandas.core.dtypes.generic import ABCSeries, ABCDataFrame
```

``not isinstance(data[x], ABCSeries)``

and change the ``DataFrame`` tests to use ``ABCDataFrame`` (and remove the inline import)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/plotting/_core.py (hunk 5 above)
Comment (MasonGallo) [reply pandas/plotting/_core.py]: > and change the DataFrame tests to use ABCDataFrame (and remove the inline import)

Not sure what you mean here? I changed the test to use `ABCDataFrame` but that doesn't seem right?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @MasonGallo nice patch! keep em coming!

File: pandas/plotting/_core.py [hunk 6]
```diff
@@ -1706,11 +1706,15 @@ def _plot(data, x=None, y=None, subplots=False,
...
+                    raise ValueError("x must be a label or position")
                 data = data.set_index(x)
 
             if y is not None:
                 if is_integer(y) and not data.columns.holds_integer():
                     y = data.columns[y]
+                elif not isinstance(data[y], Series):
```
Comment (flutefreak7) [inline pandas/plotting/_core.py]: So if I do df.plot.line(x='x', y=['y1', 'y2', 'y3']) this is now a ValueError?

**Comment by TomAugspurger** [comment]
See https://github.com/pandas-dev/pandas/issues/19699

**Pull request merged**